### ISSUE TITLE
Follow Tag (replacing “direct” score)

### ISSUE THREAD

**Issue by mirie** [forem/forem#18871]
## User Story (Who / What / Why)

As a user, I’d like to be able to more easily manage my followed tags so that I can more easily curate the content on my home feed

As an admin, I’d like to create Display Ads to help guide people here, but need confidence that the end experience is friendlier 🙂

## Designs/Details

Figma designs: [https://www.figma.com/file/nj32OuyDOR430frESjcOCM/Tag-Interaction?node-id=1105%3A599&t=LRpgegw0X7XMFKLV-4](https://www.figma.com/file/nj32OuyDOR430frESjcOCM/Tag-Interaction?node-id=1105%3A599&t=LRpgegw0X7XMFKLV-4) (see V1 section - designs show both following and new favoriting behavior so refer to screenshots below)

Currently the **Following Tags page** looks like this:

![Image](https://user-images.githubusercontent.com/896728/207964680-48399ad0-aaf9-44a2-9b4f-40584d51c3f9.png)

The updated design would change the card appearance and behavior:

- No top border color
- Remove the explicit weighting and streamline preference update interaction
- Addition of “Following” button *(starring/favoriting a tag happens in a subsequent issue)*
- To “Follow” a tag means that the weight of the tag is 1.0

![Image](https://user-images.githubusercontent.com/896728/207964729-74e452ad-b0d4-4c41-b2ff-0d2ff73779d3.png)

Currently a **Tag landing page** looks like this and there will be no change here:

![Image](https://user-images.githubusercontent.com/896728/207964763-448e0a18-8f92-4d76-9ff5-7ffac9e38fed.png)

## Definition of Done (Following)

******On****** `/dashboard/following_tags` 

- [ ]  Displays in alphabetical order all of the tags that a user follows
- [ ]  Clicking on Following button for a given Tag will unfollow the tag. On refresh of page that tag card won’t be visible

**Comment by mirie** [comment]
@pkfrank / @benhalpern :  When we move from explicit weight to just follow and unfollow behavior, will we ignore the weighting going forward on this Dashboard? I know that we will still use the weights for the feed algorithm, but wanted to double check that we simply display on `dashboard/following_tags` page as an alphabetical list.

**Comment by michael-tharrington** [comment]
So siked to hear about this coming change!

I'm wondering if there's any plan to offer folks a way to ignore tags in the wake of removing negative tag weights.

It would be super cool if we offered folks a way to ignore tags at the same time as removing the tag weighting... maybe we could run a query to make it so that if someone has negatively followed tags that those tags will be moved to ignored.

If we have to wait to a later time to offer ignoring tags as a feature, I think that's understandable, but it's clearly something that folks want. ([It's the mostly highly requested feature we have in our repo according to 👍.](https://github.com/forem/forem/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc))

**Comment by anujbhavsar96** [comment]
> I'm wondering if there's any plan to offer folks a way to ignore tags in the wake of removing negative tag weights.

@michael-tharrington Isn't `not following` the tag means ignoring it? What was the use case of negative weight?

**Comment by michael-tharrington** [comment]
Good question @anujbhavsar96!

I think it's best answered via an example.

Let's say I follow an author that writes a lot about JS, but occasionally about Rust. I love Rust but hate JS (just for this example, haha... really, I don't have an opinion)... how can I manage the things I follow to not see JS posts by that author?

If I had the ability to ignore JS, then I could follow that author and feel pretty reassured that I'd be less likely to see JS posts from them, but fairly likely to see Rust posts from them.

Let's look at another example...

What if I joined DEV, but I really don't like JS or web dev. I'm not following these tags but I still see them in my "relevant" feed often because of the sheer volume of posts tagged with #webdev and #js. I don't have any way to tell the system to stop showing me these posts in my feed. (That said, maybe I'm wrong about this... maybe simply because I'm not following these tags, I'll not see them in my feed. Can you confirm if that's the case? Anecdotally, I don't think that is the case — I don't think that I'm able to just not follow a tag and no longer see content about it... but I'd love to be proven wrong!)

Lastly, I think there's something to say for the _perception_ of quality. If we don't allow folks to ignore a tag, they feel like they don't have the ability to control what they don't see — that's not a perception that we want folks to have. Again, I know that others want this feature because if you look at our issues and sort by 👍, it's the most liked and [top 3 most commented](https://github.com/forem/forem/issues?q=is%3Aissue+is%3Aopen+sort%3Acomments-desc) request in our repo. That's a sign that folks perceive this as a necessary tool to control what they're seeing. I think we should really consider meeting folks where they are with this request.

**Comment by anujbhavsar96** [comment]
Thanks for explaining. It totally makes sense ✨ @michael-tharrington 

I think we should improve our feed in a way that the most relevant and followed tags come on top. 

In your example, if Rust and JAVA are included as tags, and rust only is followed, then you can display them slightly lower in the feed. Or some more relevant logic.

We can combine this with the idea I shared in slack about the `not interested` post. This way, users can select what kind of content they want to see while reducing the visibility of the type of content they don't want to see.

Also, we have a favorite tag option here - a favorite tag will have a higher score, so it'll come on top. In a way, we are asking users more about what they want to see. What they do not want to see can be achieved via the `not interested` feature.

In short, we can say `not interested` is kind of like a negative tag score; Follow is mid-level, and favorite is a positive tag score. This will be much better interaction than manual tag weight input.

**Comment by michael-tharrington** [comment]
Good stuff @anujbhavsar96!

> I think we should improve our feed in a way that the most relevant and followed tags come on top.

I def agree with ya here, but I also think that to be successful with that, we need to not show folks the things they don't wanna see.

That said, I also agree that the idea you proposed for `not interested` goes a long way to solve that problem — giving folks control over the things they don't wanna see.

I'm still a proponent of offering folks the ability to ignore a tag, but I think I've made all the points that I can in advocating for this feature. If ultimately we decide not to offer it, I do think letting folks tell us they're `not interested` in a post is a decent compromise (though I also think the 2 features could work hand-in-hand).

Favorite tag is a cool idea... maybe we could extend that to favorite author, org, and podcast also. Just a thought though, I don't feel super strongly about adding that. I do think tag is most important in the favoriting context.

And I absolutely agree that negative following weights have gotta go, haha... I don't really see `not interested` in a post as completely filling that gap, I see it more as a way to handle things in the present versus handling future things. In other words, if I tell the system I'm not interested in a post, it would remove the post from my feed, but it's still kinda fuzzy to me if/how that would affect what I see in my feed going forward. Also, would folks be able to go somewhere to see the `not interested` choices they've made? For instance, I regularly read things in the Google feed of my Android phone and from the feed, I can click on something that's called "Manage your interests" where I'm able to see "your interests" and "not interested" — clicking into either gives me control over these two lists. Anyway, as we continue brainstorming implementation details on this and talk through it, I bet this will all become clearer.
